### DISCUSSION TITLE
nextflow and carpet

### DISCUSSION THREAD

**Discussion by jagedn** [jerolba/parquet-carpet#34] [Show and tell]
Nextflow is a DSL that enables scalable and reproducible scientific workflows using software containers. 

It's based in ApacheGroovy and is extensible via plugins. Out of the box it contains a core implementation to read CSV files so every line is emitted to processes as a Map

Today I've created (and hopefully published soon) a new plugin, nf-parquet, and using Carpet now pipelines are able to read/write parquet files. You can read more about the plugin at https://edn-es.github.io/nf-parquet/

Thanks for this lib @jerolba

**Comment by jerolba** [comment]
Awesome Jorge! 

I didn't know about Nextflow. I must read about it to see if it's useful for my work :) 

In the documentation is see that "it requires Java version 19 as a minimum", why Java 19 and not 17? 
17 is the lower LTS version that supports records.

Thanks for your support on Carpet!
Jero

**Reply by jagedn** [reply]
Surely it works also with java 17 (I'll check) but my tests where failing due some issues in Nextflow + Gradle

Nextflow is very popular in the bioinformatics sector but it's not restricted to it 

I wrote a couple of introduction posts:

https://jorge.aguilera.soy/tags/nextflow.html

**Comment by jagedn** [comment]
I would like to share that Nextflow's Parquet plugin is gaining popularity in the Bioinformatic community reaching 1.8K downloads

In the roadmap is and upgrade to a more recent Carpet version but current version is working like a charm, good job @jerolba 

https://registry.nextflow.io/plugins/nf-parquet/trends

<img width="2367" height="442" alt="imagen" src="https://github.com/user-attachments/assets/f0a4ae85-5028-465a-8153-2784616e905d" />
